packages/zoe/src/zoeService/instanceAdminStorage.js

```diff
@@ -1,49 +1,284 @@
 import { E } from '@endo/eventual-send';
-import { makeWeakStore } from '@agoric/store';
+import {
+  canBeDurable,
+  makeScalarBigSetStore,
+  provideDurableWeakMapStore,
+  vivifyKindMulti,
+  vivifyFarClassKit,
+  M,
+  provide,
+} from '@agoric/vat-data';
+import { makeZoeSeatAdminFactory } from './zoeSeat.js';
+import { defineDurableHandle } from '../makeHandle.js';
+import { InstanceAdminI, InstanceHandleShape } from '../typeGuards.js';
 
-/**
- *
- */
-export const makeInstanceAdminStorage = () => {
-  /** @type {WeakStore<Instance,InstanceAdmin>} */
-  const instanceToInstanceAdmin = makeWeakStore('instance');
+const { quote: q, details: X } = assert;
 
-  /** @type {import('./utils.js').GetPublicFacet} */
-  const getPublicFacet = async instanceP =>
-    E.when(instanceP, instance =>
-      instanceToInstanceAdmin.get(instance).getPublicFacet(),
-    );
+const InstanceAdminStorageShape = M.interface('InstanceAdminStorage', {
+  getPublicFacet: M.call(M.eref(InstanceHandleShape)).returns(M.promise()),
```

Comment:
```suggestion
  getPublicFacet: M.callWhen(M.await(InstanceHandleShape)).returns(M.promise()),
```

Likely, searching for `M.eref(` will identify many such opportunities. But it is only a valid refactor if the `E.when` is there and wraps the entire rest of the function.

Reply:
> But it is only a valid refactor if the E.when is there and wraps the entire rest of the function.

Or if there should have been an `E.when()` in the function, but it was missing!

Reply:
Cool!

Looks done. Resolving.